pandas/core/dtypes/cast.py

```diff
@@ -942,6 +942,32 @@ def coerce_indexer_dtype(indexer, categories):
     return ensure_int64(indexer)
 
 
+def _find_level() -> int:
+    """
+    Find the appropriate stacklevel with which to issue a warning for astype.
```

Comment:
i would move this elsewhere, maybe to pandas.util._validation (or similar).

maybe just pass the names in as needed e.g. L958 (or hard-code ok too)

Reply:
maybe util._exceptions?

Reply:
sure

Reply:
updated + green